src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabServerPullRequestDecorator.java

```diff
@@ -146,11 +146,11 @@ public void decorateQualityGateStatus(AnalysisDetails analysis, UnifyConfigurati
                 }
             }
 
-            QualityGate.Condition newCoverageCondition = analysis.findQualityGateCondition(CoreMetrics.NEW_COVERAGE_KEY)
-                    .orElseThrow(() -> new IllegalStateException("Could not find New Coverage Condition in analysis"));
-            String coverageValue = newCoverageCondition.getStatus().equals(QualityGate.EvaluationStatus.NO_VALUE) ? "0" : newCoverageCondition.getValue();
-
-
+            String coverageValue = analysis.findQualityGateCondition(CoreMetrics.NEW_COVERAGE_KEY)
+                .map(
+                    c ->  c.getStatus().equals(QualityGate.EvaluationStatus.NO_VALUE) ? "0" : c.getValue()
+                ).orElse("0")
+            ;
```

Comment:
Can this not be expressed without ternary operations as the following?
```suggestion
            String coverageValue = analysis.findQualityGateCondition(CoreMetrics.NEW_COVERAGE_KEY)
                .filter(c -> ! c.getStatus().equals(QualityGate.EvaluationStatus.NO_VALUE))
                .map(QualityGate.Condition::getValue)
                .orElse("0");
```